google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TableResult.java

```diff
@@ -46,24 +48,35 @@ public JobId getJobId() {
     return jobId;
   }
 
+  public String getQueryId() {
+    return queryId;
+  }
+
   /**
    * If {@code schema} is non-null, {@code TableResult} adds the schema to {@code FieldValueList}s
    * when iterating through them. {@code pageNoSchema} must not be null.
    */
   @InternalApi("Exposed for testing")
-  public TableResult(Schema schema, long totalRows, Page<FieldValueList> pageNoSchema) {
+  public TableResult(
```

Comment:
Rather than solely extending the existing constructor, should we just provide an second signature or a builder?  It would mean you can avoid having to touch all the callsites where queryId isn't relevant for construction.

Reply:
As discussed offline. I think creating a builder would be the preferred option. I'll follow up with another PR to switch to using a builder.